server/database/migrations/2016_07_25_203547_mark_note_replies_as_notes.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+use App\Models\Annotation;
+
+class MarkNoteRepliesAsNotes extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $annotations = DB::table('annotations')->whereNotNull('data')->get();
+        foreach ($annotations as $annotation) {
+            $data = json_decode($annotation->data, true);
+
+            if (empty($data['old_permalink_type']) || $data['old_permalink_type'] !== 'annsubcomment') {
```

Comment:
In general, these should be part of the where clause instead.  You could, actually, do this whole thing with one single update query instead of having to loop over results.